source/LocationFilter.cpp

```diff
@@ -470,7 +470,7 @@ void LocationFilter::LoadChild(const DataNode &child)
 	else if(key == "category" && child.Size() >= 2 + isNot)
 	{
 		// Ship categories cannot be combined in an "and" condition.
-		static const set<string> allowed(Ship::CATEGORIES.begin(), Ship::CATEGORIES.end());
+		static const set<string> allowed(GameData::Categories("ship").begin(), GameData::Categories("ship").end());
```

Comment:
I do have a concern here that file load order may cause issues. The first specification of a ship category (e.g. from a job or mission in the base game or earlier-loaded plugin) as a condition in a `LocationFilter` will result in the creation and finalization of this variable `allowed`, and it will persist while the game runs. Any further additions to the category list will not be reflected in the following code block.

I.e.
```
Base -> categories.txt
Base -> jobs.txt
...
Plugin 1 -> so many missions.txt
Plugin 2 -> boarding jobs.txt
Plugin 2 -> ship categories.txt
```

Any categories in `ship categories.txt` are not even known when `boarding jobs.txt` is parsed. And even if they were known, they would still not be in `allowed` if any preceding file reached this line.

Reply:
Confirmed that this is a bug.